### PR TITLE
feat(crc): counterparty contact info

### PR DESCRIPTION
**Pull request by VilliGunn**
# \<Description\>

https://app.asana.com/0/1199733748699731/1200184411501168

## What

- New contact info screen with email and phone number fields so we can notify parent B about the application in progress.'
- Updated `InputController` to be able to hook into the `onChange` event of Numeric fields

## Why

Parent A should fill out an email or phone number field so we can notify parent B that there is an application waiting their approval on island.is

## Screenshots / Gifs

Screenshot 2021-04-20 at 22 58 16

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/children-residence-change/src/fields/ContactInfo/index.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  const counterPartyError = errors?.counterParty as string
+  const isCounterPartyError = typeof counterPartyError === 'string'
+  const counterPartyEmailError = errors?.counterParty?.email
+  const counterPartyPhoneError = errors?.counterParty?.phoneNumber
+  const applicant = nationalRegistry.data
+  const childResidenceInfo = childrenResidenceInfo(applicant, answers)
+
+  return (
+    <>
+      <Text marginTop={3}>
```
Comment (VilliGunn) [inline libs/application/templates/children-residence-change/src/fields/ContactInfo/index.tsx:47]: I might refactor and extract some of this to a seperate component this when working on this task: https://app.asana.com/0/1199733748699731/1200222247492226

**Review by Rikkiabb — COMMENTED** [review]
Couple of comments

File: libs/application/template-api-modules/src/lib/modules/templates/children-residence-change/children-residence-change.service.ts [hunk 2]
```diff
@@ -146,15 +146,14 @@ export class ChildrenResidenceChangeService {
...
-    if (counterParty?.email) {
+    if (counterParty.email) {
       await this.sharedTemplateAPIService.sendEmail(
         transferRequestedEmail,
         (application as unknown) as Application,
       )
     }
 
-    if (counterParty?.phoneNumber) {
+    if (counterParty.phoneNumber) {
```
Comment (Rikkiabb) [inline libs/application/template-api-modules/src/lib/modules/templates/children-residence-change/children-residence-change.service.ts:156]: Not in this pr, but we should add unit tests for this service. I'll add a ticket for it.

**Review by Rikkiabb — APPROVED** [review]
👌

**Review by jeremybarbet — APPROVED** [review]
LGTM 👍

**Pull request merged**